deployments/helm/templates/configmap.yaml

```diff
@@ -15,5 +15,17 @@ metadata:
   {{- end }}
 data:
   beyla-config.yml: |
+    {{- if eq .Values.preset "network" }}
+    {{- if not .Values.config.data.network }}
+    network:
+      enable: true
+    {{- end }}
+    {{- else }}
+    {{- if not .Values.config.data.discovery }}
+    discovery:
+      services:
+        - k8s_namespace: .
+    {{- end }}
+    {{- end }}
```

Comment:
I don't know if there is a cleaner way to do that. The rationale behind this is that we want either network or discovery sections depending on the selected preset, but not both.

The values.yaml file does not accept template control flows.

We don't want to print the network or discovery if they have been already defined by the user.